### ISSUE TITLE
option: show/hide scrollbars

### ISSUE THREAD

**Issue by s-bl** [The-Compiler/qutebrowser#541]
would be nice to have an option to show/hide scrollbars.

**Comment by The-Compiler** [comment]
Currently hiding scrollbars is implemented as an user stylesheet. If you prefer to show them, you can do `:set ui user-stylesheet ""`.

After #135 (*support for multiple user-stylesheet files*) I also plan to add an option for that, so I'll close this one.

**Comment by Veraellyunjie** [comment]
@The-Compiler Please reconsider this issue.

In `qutebrowser` `1.6.1-2` (`Debian Buster`) scrollbars used to be on by default, now I have installed `qutebrowser` `2.3.1` (`OpenBSD 7.0`) where they are on-hover.

`:set ui user-stylesheet ""` returns an error `set: Unrecognized arguments:`;
and in `qute://settings/` the `content.user_stylesheets` is empty.

How do I get back those good old scrollbars which are visible whenever the content doesn't fit into the viewport?

**Comment by The-Compiler** [comment]
@Veraellyunjie Things happen to be different than they were 7 years ago (!), and the on-hover scrollbars are unrelated to this issue entirely...

From [the changelog](https://github.com/qutebrowser/qutebrowser/blob/master/doc/changelog.asciidoc#v1130-2020-06-26):

> `scrolling.bar` supports a new `overlay` value to show an overlay scrollbar, which is now the default. On unsupported configurations (on Qt < 5.11, with QtWebKit or on macOS), the value falls back to `when-searching` or `never` (QtWebKit).

So you might want to set it to one of those values instead (followed by a `:restart`).

**Comment by Veraellyunjie** [comment]
@The-Compiler
The `scrolling.bar` `never` that you mentioned made me try `always` and it does work, now those scrollbars are visible as I want them to be. (`OpenBSD` `7.0`, `qutebrowser` `2.3.1`, `QtWebEngine`, `Qt5` `5.15.4`.)
Thank you.

---

BTW, it would be great (and I ask you for it) if the description comments on `qute://settings/` page also contained available values, like the `tabs.title.format` description does. `scrolling.bar` has none, for instance, so it is only after your answer and a bit of antonym logic that I could make use of the option field. And `content.proxy`: having set it to `socks://localhost:9050`, when I need to disable proxy I have already forgotten the default value (`system`) and there is no reminder. And so on…

---

my `--verbose` mode `on`:

> on-hover scrollbars are unrelated to this issue entirely...

Well, from the user perspective it seems like the same group of options, IMHO

> Things happen to be different than they were 7 years ago (!)

Well, the desktop itself and desktop needs have not changed, IMHO

**Comment by The-Compiler** [comment]
> The `scrolling.bar` `never` that you mentioned made me try `always` and it does work, now those scrollbars are visible as I want them to be. (`OpenBSD` `7.0`, `qutebrowser` `2.3.1`, `QtWebEngine`, `Qt5` `5.15.4`.) Thank you.

You're welcome!

> BTW, it would be great (and I ask you for it) if the description comments on `qute://settings/` page also contained available values, like the `tabs.title.format` description does.

> `scrolling.bar` has none, for instance, so it is only after your answer and a bit of antonym logic that I could make use of the option field. And `content.proxy`: having set it to `socks://localhost:9050`, when I need to disable proxy I have already forgotten the default value (`system`) and there is no reminder. And so on…

That's already the case on git master, at least for settings with a fixed set of values, thanks to @twigleingrid - see #6818.

Note that you can use the setting completion you get after `:set content.proxy ` as well as `:help content.proxy`.

> > on-hover scrollbars are unrelated to this issue entirely...
> 
> Well, from the user perspective it seems like the same group of options, IMHO

This is a 7 year old issue about having the possibility to show/hide the scrollbars. You were asking about a much more recent change about showing "full" rather than the overlay scrollbars. Sure, they both happen to be about scrollbars, but it's still unclear what you're asking me to reconsider, given that this issue is about a different thing that you seem to be talking about.

> > Things happen to be different than they were 7 years ago (!)
> 
> Well, the desktop itself and desktop needs have not changed, IMHO

What I was saying is that you can't usually take qutebrowser commands I mentioned 7 years ago and expect them to still work the same way on a current version.
